### ISSUE TITLE
Bad `pdflatex` executable: `/usr/bin/pdflatex`

### ISSUE THREAD

**Issue by bs89** [textext/textext#373]
I followed the instructions to install textext on my linux machine. When I ran `setup.py` it was able to automatically find my pdflatex installation (texlive) and the installation was successful. However, when I run the extension in inkscape it gives and error message similar to the one below and it basically says that it cannot find my pdflatex. Then I tried specifying the pdflatex path explicitly when installing the package and I got the message below. For some reason it does not like my pdflatex executable (which is working fine though), so it still wants to search for it. At the same time, I can see that /usr/bin is in the path below where it is supposed to find it.

Any idea what is going on? Thanks for your help :)

```
[TexText][DEBUG   ]: First run of TexText. Enforcing DEBUG mode.
[TexText][DEBUG   ]: TexText initialized
[TexText][DEBUG   ]: TexText version = '1.8.2' (md5sum = 30841cf359fabd5cacc4ba8a123a344e)
[TexText][DEBUG   ]: platform.system() = 'Linux'
[TexText][DEBUG   ]: platform.release() = '5.19.0-76051900-generic'
[TexText][DEBUG   ]: platform.version() = '#202207312230~1663791054~22.04~28340d4 SMP PREEMPT_DYNAMIC Wed S'
[TexText][DEBUG   ]: platform.machine() = 'x86_64'
[TexText][DEBUG   ]: platform.uname() = uname_result(system='Linux', node='pop-os', release='5.19.0-76051900-generic', version='#202207312230~1663791054~22.04~28340d4 SMP PREEMPT_DYNAMIC Wed S', machine='x86_64')
[TexText][DEBUG   ]: platform.mac_ver() = ('', ('', '', ''), '')
[TexText][DEBUG   ]: sys.executable = '/usr/bin/python3'
[TexText][DEBUG   ]: sys.version = '3.9.9 (main, Nov 10 2011, 15:00:00) \n[GCC 11.3.0]'
[TexText][DEBUG   ]: os.environ = environ({'XDG_RUNTIME_DIR': '/run/user/1000', 'container': 'flatpak', 'ALSA_CONFIG_DIR': '/usr/share/alsa', 'ALSA_CONFIG_PATH': '/usr/share/alsa/alsa-flatpak.conf', 'AT_SPI_BUS_ADDRESS': 'unix:path=/run/flatpak/at-spi-bus', 'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/flatpak/bus', 'DESKTOP_SESSION': 'pop', 'DISPLAY': ':99.0', 'FLATPAK_ID': 'org.inkscape.Inkscape', 'FLATPAK_SANDBOX_DIR': '/home/username/.var/app/org.inkscape.Inkscape/sandbox', 'GDMSESSION': 'pop', 'GIO_LAUNCHED_DESKTOP_FILE': '/home/username/.local/share/flatpak/exports/share/applications/org.inkscape.Inkscape.desktop', 'GIO_LAUNCHED_DESKTOP_FILE_PID': '7641', 'GI_TYPELIB_PATH': '/app/lib/girepository-1.0', 'GJS_DEBUG_OUTPUT': 'stderr', 'GJS_DEBUG_TOPICS': 'JS ERROR;JS LOG', 'GNOME_DESKTOP_SESSION_ID': 'this-is-deprecated', 'GNOME_SHELL_SESSION_MODE': 'pop', 'GPG_AGENT_INFO': '/run/user/1000/gnupg/S.gpg-agent:0:1', 'GST_PLUGIN_SYSTEM_PATH': '/app/lib/gstreamer-1.0:/usr/lib/extensions/gstreamer-1.0:/usr/lib/x86_64-linux-gnu/gstreamer-1.0', 'GTK_IM_MODULE': 'ibus', 'GTK_MODULES': 'gail:atk-bridge:appmenu-gtk-module', 'HOME': '/home/username', 'INVOCATION_ID': '5adb5992b4754dc5a5fc967632f220cf', 'JOURNAL_STREAM': '8:39495', 'LANG': 'en_US.UTF-8', 'LD_LIBRARY_PATH': '', 'LOGNAME': 'username', 'MANAGERPID': '2026', 'PATH': '/app/bin:/app/bin:/usr/bin', 'PS1': '[📦 $FLATPAK_ID \\W]\\$ ', 'PWD': '/home/username', 'QT_ACCESSIBILITY': '1', 'QT_IM_MODULE': 'ibus', 'SESSION_MANAGER': 'local/pop-os:@/tmp/.ICE-unix/2232,unix/pop-os:/tmp/.ICE-unix/2232', 'SHELL': '/bin/sh', 'SHLVL': '0', 'SSH_AGENT_LAUNCHER': 'gnome-keyring', 'SSH_AUTH_SOCK': '/run/user/1000/keyring/ssh', 'SYSTEMD_EXEC_PID': '2258', 'USER': 'username', 'USERNAME': 'username', 'WINDOWPATH': '2', 'XAUTHORITY': '/run/flatpak/Xauthority', 'XDG_CACHE_HOME': '/home/username/.var/app/org.inkscape.Inkscape/cache', 'XDG_CONFIG_DIRS': '/app/etc/xdg:/etc/xdg', 'XDG_CONFIG_HOME': '/home/username/.var/app/org.inkscape.Inkscape/config', 'XDG_CURRENT_DESKTOP': 'pop:GNOME', 'XDG_DATA_DIRS': '/app/share:/usr/share:/usr/share/runtime/share:/run/host/user-share:/run/host/share:/app/share/inkscape', 'XDG_DATA_HOME': '/home/username/.var/app/org.inkscape.Inkscape/data', 'XDG_MENU_PREFIX': 'gnome-', 'XDG_SESSION_CLASS': 'user', 'XDG_SESSION_DESKTOP': 'pop', 'XDG_SESSION_TYPE': 'x11', 'XMODIFIERS': '@im=ibus', '__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS': '/etc/egl/egl_external_platform.d:/usr/lib/x86_64-linux-gnu/GL/egl/egl_external_platform.d:/usr/share/egl/egl_external_platform.d', 'PYTHONPATH': '/home/username/.var/app/org.inkscape.Inkscape/config/inkscape/extensions:/app/share/inkscape/extensions:/app/share/inkscape/extensions/inkex/deprecated-simple', 'PYTHONIOENCODING': 'UTF-8', 'INKSCAPE_PROFILE_DIR': '/home/username/.var/app/org.inkscape.Inkscape/config/inkscape', 'DOCUMENT_PATH': '', 'INKEX_GETTEXT_DOMAIN': 'inkscape', 'INKEX_GETTEXT_DIRECTORY': '/app/share/locale'})
[TexText][WARNING ]: Bad `pdflatex` executable: `/usr/bin/pdflatex`
[TexText][WARNING ]: Fall back to automatic detection of `pdflatex`
[TexText][CRITICAL]: + [Fail] TexText requirements
[TexText][SUCCESS ]: /-and-* [Succ] Detect inkscape>=1.0
[TexText][SUCCESS ]: |      inkscape=1.2 is found
[TexText][CRITICAL]: /-and-+ [Fail] Detect *latex
[TexText][CRITICAL]: |      Please follow installation instructions at 
[TexText][CRITICAL]: |             [94m[4mhttps://textext.github.io/textext/install/linux.html#linux-install-preparation[0m
[TexText][CRITICAL]: |     /--or-* [Fail] `pdflatex` is NOT found in PATH
[TexText][CRITICAL]: |     |      Please follow installation instructions at 
[TexText][CRITICAL]: |     |             [94m[4mhttps://textext.github.io/textext/install/linux.html#linux-install-preparation[0m
[TexText][CRITICAL]: |     |      If pdflatex is installed in custom location, specify it via 
[TexText][CRITICAL]: |     |             --pdflatex-executable=<path-to-pdflatex>
[TexText][CRITICAL]: |     |      and run setup.py again
[TexText][CRITICAL]: |     /--or-* [Fail] `lualatex` is NOT found in PATH
[TexText][CRITICAL]: |     |      Please follow installation instructions at 
[TexText][CRITICAL]: |     |             [94m[4mhttps://textext.github.io/textext/install/linux.html#linux-install-preparation[0m
[TexText][CRITICAL]: |     |      If lualatex is installed in custom location, specify it via 
[TexText][CRITICAL]: |     |             --lualatex-executable=<path-to-lualatex>
[TexText][CRITICAL]: |     |      and run setup.py again
[TexText][CRITICAL]: |     /--or-* [Fail] `xelatex` is NOT found in PATH
[TexText][CRITICAL]: |            Please follow installation instructions at 
[TexText][CRITICAL]: |                   [94m[4mhttps://textext.github.io/textext/install/linux.html#linux-install-preparation[0m
[TexText][CRITICAL]: |            If xelatex is installed in custom location, specify it via 
[TexText][CRITICAL]: |                   --xelatex-executable=<path-to-xelatex>
[TexText][CRITICAL]: |            and run setup.py again
[TexText][SUCCESS ]: /-and-+ [Succ] Detect GUI library
[TexText][SUCCESS ]:       /--or-* [Succ] GTK3 is found
[TexText][SUCCESS ]:       /--or-* [Succ] TkInter is found
[TexText][ERROR   ]: TexText requirements are not met. Please follow instructions https://textext.github.io/textext/
```

**Comment by jcwinkler** [comment]
Have you installed Inkscape via snap? In that case you won't be able to call pdflatex from within Inkscape since it is running in a sandbox. You need to install it using the package manager of your Linux distribution.

**Comment by bs89** [comment]
No, used `sudo apt install inkscape` as recommended by the installation guide.

**Comment by jcwinkler** [comment]
Hmm, on which Linux distro are you?

Does the Inkscape built-in basic Latex formula editor work?

**Comment by bs89** [comment]
I didn't know there was a basic Latex formula editor. It turned out that it was not working and I figured the reason why. I have to start inkscape from the command line because only then the path to its extensions are configured correctly. I've started it from the launcher before and did not know that there would be a difference. Lesson learned!

Now everything is working just fine! :) Many thanks for the useful hints!
